src/ws/cockpit-session-client-certificate.c

```diff
@@ -333,8 +319,23 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  size_t my_cgroup_length;
+  char *my_cgroup = read_proc_self_cgroup (&my_cgroup_length);
+  if (my_cgroup == NULL)
+    {
+      warnx ("Could not determine cgroup of this process");
+      return NULL;
+    }
+  if (strncmp (cert_pem, my_cgroup, my_cgroup_length) != 0)
```

Comment:
This works because both strings are expected to include the final `\n`, and because cgroups can't contain newlines. A comment for that wouldn't hurt here.